### ISSUE TITLE
Add APIs for waiting on events and sending events

### ISSUE THREAD

**Issue by sujayakar** [get-convex/workflow#25]

**Comment by shiftlabs1** [comment]
@sujayakar , is this similar to message passing in temporal which provides the ability to update workflow state from outside the workflow and to respond to state change within the work flow , like message passing in temporal ? I just asked about this on discord . In temporal this message passing https://docs.temporal.io/develop/typescript/message-passing . If yes then this is very very much needed to make this project usable

**Comment by sujayakar** [comment]
yep, exactly! I can take a look at implementing this over the next few weeks. I'd also be curious what other features are needed for your use case to make this library usable -- are there others than message passing?

**Comment by shiftlabs1** [comment]
I will look through my other implementations using temporal and restate and let you know/ post here as new issues. The really cool part of the convex workflow , is how close it is to the data , when compared to the other offerings .
